alyx/buffalo/admin.py

```diff
@@ -287,6 +288,26 @@ class SessionTaskInline(nested_admin.NestedTabularInline):
     inlines = [SessionDataNestedsetInline]
     ordering = ("task_sequence",)
 
+    def formfield_for_foreignkey(self, db_field, request, **kwargs):
+        subject = request.GET.get("subject", None)
+        if db_field.name == "task":
+            sessions = BuffaloSession.objects.filter(
+                subject__id=subject
+            ).order_by('-start_time')
```

Comment:
Here you can add .first() to get just the one record you need, then sessions variable will be singular because it will have just the last session if any